### PR TITLE
Restructure HSI_Toolkit for packaging

### PR DESCRIPTION
**Pull request by HaleySchuhl**
Restructure the repo format to allow for a namespace package called hsi_toolkit. Added required files such as requirements, manifest, setup.py, and setup.cfg files. 

Removed parts of code that I found that is changing the path. 

[x] Tested compatibility with upload on PyPI
[x] Tested all functions imported by creating a new conda environment, installing hsi_toolkit, and then used the `help` function to check contents of all sub-packages (anomaly_detectors, classifiers, dim_reduction, endmember_extraction, signature_detectors, and util)

*Additional Context/Discussion* 
When we discussed packaging up these tools there was talk about only packaging up well tested tools. At the time of opening this PR, all functions that were available on the hsi_toolkit_py repo are imported with hsi_toolkit. However, Noah and I discussed how to efficiently update this once we receive a list of which functions are ready for primetime we can leave these functions where they are and move all beta-functions into a DEV sub-package that is not important with the package but this would still make this raw code available on this repo! 

[ ] Add "under development" sub-package

### CODE REVIEW

**Comment by HaleySchuhl** [comment]
@susanmeerdink just wanted to confirm that `hsi_toolkit.signature_detectors.ftmf_detector` should go into the dev branch (also, is that a different name for the in progress sub-package that you all would prefer?). In your email you listed ftmf_statistic but I didn't see a function by this name. I came across one other question that comes down to your preference. Would you like all the dev functions sit within that sub-package or instead I could create folders inside the dev sub-package to mimic the tested python package structure. I can add these if you'd like!

**Comment by susanmeerdink** [comment]
@HaleySchuhl Sorry for the typo! Yes I do mean hsi_toolkit.signature_detectors.ftmf_detector instead of ftmf_statistic. Please keep it as ftmf_detector instead of ftmf_statistic. I'm sorry but could you elaborate on or use examples of what my two options are for the in-progress functions? I'm not sure I understand what the two restructuring options are.

**Comment by HaleySchuhl** [comment]
Option 1 (this is how the PR is currently): 

HSI_toolkit 
    >anomaly_detectors sub-package
    >classifiers sub-package
    >dev sub-package
        >>ALL IN PROGRESS FUNCTION FILES
    >dim_reduction 
    >endmember_extraction
    >signature_detectors
    >util 

Option 2 : 

HSI_toolkit 
    >anomaly_detectors sub-package
    >classifiers sub-package
    >dev sub-package
    >anomaly_detectors sub-package
    >classifiers sub-package
    >dev sub-package
       >> folder of anomaly_detectors 
        >>folder of classifiers 
        >>folder of sub-package
        >>folder of anomaly_detectors 
        >>folder of classifiers sub-package
        >>folder of dim_reduction
        >>folder of dim_reduction 
        >>folder of endmember_extraction
        >>folder of signature_detectors
        >>folder of util 
    >dim_reduction 
    >endmember_extraction
    >signature_detectors
    >util

**Comment by susanmeerdink** [comment]
@HaleySchuhl Thank you for the clarification. We would like to go with Option 1, but within the ALL IN PROGRESS folder have subfolders for each category (anomaly, classifiers, dim reduction, etc). Does that make sense?

**Comment by HaleySchuhl** [comment]
I think this PR should be good to go once the conflicts are resolved! Just let me know if you have any comments/suggestions/questions! 😄

File: .idea/encodings.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nfahlgren) [inline .idea/encodings.xml:1]: Some more PyCharm files snuck in.

File: .idea/hsi_toolkit_PR.iml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nfahlgren) [inline .idea/hsi_toolkit_PR.iml:1]: Some more PyCharm files snuck in.

File: .idea/markdown-navigator.xml [hunk 3]
```diff
@@ -0,0 +1,86 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nfahlgren) [inline .idea/markdown-navigator.xml:1]: Some more PyCharm files snuck in.

File: .idea/markdown-navigator/profiles_settings.xml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<component name="MarkdownNavigator.ProfileManager">
```
Comment (nfahlgren) [inline .idea/markdown-navigator/profiles_settings.xml:1]: Some more PyCharm files snuck in.

File: .idea/misc.xml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nfahlgren) [inline .idea/misc.xml:1]: Some more PyCharm files snuck in.

File: .idea/modules.xml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nfahlgren) [inline .idea/modules.xml:1]: Some more PyCharm files snuck in.

File: .idea/vcs.xml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nfahlgren) [inline .idea/vcs.xml:1]: Some more PyCharm files snuck in.

File: requirements.txt [hunk 8]
```diff
@@ -0,0 +1,3 @@
+numpy
+scipy
+sklearn
```
Comment (nfahlgren) [inline requirements.txt:3]:
```suggestion
scikit-learn
pca
cvxopt
```

**Review by nfahlgren — CHANGES_REQUESTED** [review]
@HaleySchuhl I marked some PyCharm files that snuck in and updated the requirements.txt file a bit

**Comment by nfahlgren** [comment]
@HaleySchuhl I would move all the demo scripts to a common demo folder in the main repo I think

**Comment by nfahlgren** [comment]
@HaleySchuhl the file `hsi_toolkit/dev/endmember_extraction/__init__.py` should have:

```python
from hsi_toolkit.endmember_extraction.VCA import *
```

**Comment by nfahlgren** [comment]
@HaleySchuhl I did some testing with SPICE and I think we just need to add `SPICE_HSI` to the `requirements.txt` file and in the file `hsi_toolkit/dev/endmember_extraction/__init__.py` we would add `import SPICE`

**Pull request closed without merging**